External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp

```diff
@@ -94,6 +94,62 @@ static uint64_t AtomicFetchNeg(uint64_t *Addr) {
 
   return Expected;
 }
+
+static uint8_t AtomicCompareAndSwap8(uint8_t expected, uint8_t desired, uint8_t *addr)
```

Comment:
Looks like this could be turned into a template so we don't need to duplicate the same code three times here.

For the ARM implementations, we can use template specialization, i.e.
```cpp
template<typename T>
static T AtomicCompareAndSwap(T expected, T desired, T *addr);

template<>
uint16_t AtomicCompareAndSwap(uint16_t expected, uint16_t desired, uint16_t* addr) {
    // Impl goes here   
}
```

Reply:
Replaced x86_64 impl with a template. ARM64 uses template specialization.